### PR TITLE
some

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                        .prepareStatement(sqlQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.executeUpdate();
+            ResultSet resultset = preparedStatement.getGeneratedKeys();
+            while (resultset.next()) {
+                manufacturer.setId(resultset.getObject(1, Long.class));
+            }
+            preparedStatement.close();
+            connection.close();
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: If you use "try with resources", you don't need to close them after

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery);) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultset = preparedStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            if (resultset.next()) {
+                manufacturer = createManufacturer(resultset);
+            }
+            preparedStatement.close();
+            connection.close();
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: same

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        String query = "SELECT * FROM manufacturer WHERE is_deleted = false;";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultset = preparedStatement.executeQuery();
+            while (resultset.next()) {
+                manufacturers.add(createManufacturer(resultset));
+            }
+            preparedStatement.close();
+            connection.close();
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: same

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        String query = "UPDATE manufacturer SET manufacturer_name=?, manufacturer_country=?"
+                + " WHERE manufacturer_id =?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
+            connection.close();
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: same

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    public boolean delete(Long id) {
+        String sqlQuery = "UPDATE manufacturer"
+                + " SET is_deleted = TRUE"
+                + " WHERE manufacturer_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery)) {
+            preparedStatement.setLong(1, id);
+            int updatesMade = preparedStatement.executeUpdate();
+            preparedStatement.close();
+            connection.close();
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: same

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            return updatesMade > 0;
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't delete manufacturer by id " + id, ex);
+        }
+    }
+
+    private Manufacturer createManufacturer(ResultSet resultset) throws SQLException {
+        Long id = resultset.getObject("manufacturer_id", Long.class);
+        String name = resultset.getObject("manufacturer_name", String.class);
+        String country = resultset.getObject("manufacturer_country", String.class);
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: here, you can use getString();

File: src/main/java/taxi/util/ConnectionUtil.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Properties;
+
+public class ConnectionUtil {
+    static {
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Can't found MySQL driver", e);
+        }
+    }
```
Comment (MykolaFatkullin) [inline src/main/java/taxi/util/ConnectionUtil.java]: You can safely delete this static initialization. But it is not necessary

**Review by MykolaFatkullin — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 1 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: Really, I think same,but  in checklist "Best practices with closing Connections and/or PreparedStatements...". Ok, i agree.

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 2 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 3 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 4 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 5 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/taxi/util/ConnectionUtil.java (hunk 7 above)
Comment (Sashk52) [reply src/main/java/taxi/util/ConnectionUtil.java]: Done

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 6 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: Here I just follow checklist recomendation:
[https://prnt.sc/wo1ydy](url)

**Review by MykolaFatkullin — APPROVED** [review]

File: .idea/compiler.xml [hunk 8]
```diff
@@ -7,6 +7,7 @@
         <sourceTestOutputDir name="target/generated-test-sources/test-annotations" />
         <outputRelativeToContentRoot value="true" />
         <module name="git-hw-dao-three" />
+        <module name="mate-academy-github-hw" />
```
Comment (boroda4436) [inline .idea/compiler.xml]: The `.idea` folder should be not added to GitHub!

Comment (boroda4436) [reply .idea/compiler.xml]: Please remove it from GitHub and add it to .gitignore.

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+import taxi.dao.ManufacturerDao;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String sqlQuery = "INSERT INTO manufacturer (manufacturer_name, manufacturer_country)"
```
Comment (boroda4436) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]:
```suggestion
        String sqlQuery = "INSERT INTO manufacturer (name, country)"
```
You already know it is a manufacturer

File: src/main/resources/init_db.sql [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+CREATE TABLE `taxi_service`.`manufacturer` (
+                                               `manufacturer_id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+                                               `manufacturer_name` VARCHAR(225) NOT NULL,
+                                               `manufacturer_country` VARCHAR(225) NOT NULL,
+                                               `deleted` VARCHAR(5) NOT NULL,
+                                               PRIMARY KEY (`manufacturer_id`))
+    ENGINE = InnoDB
+DEFAULT CHARACTER SET = utf8;
+    CHANGE COLUMN `deleted` `deleted` VARCHAR(5) NULL DEFAULT 'FALSE' ;
+    CHANGE COLUMN `deleted` `is_deleted` VARCHAR(5) NULL DEFAULT 'FALSE' ;
```
Comment (boroda4436) [inline src/main/resources/init_db.sql]: one empty line at the end of the file is required

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 9 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: done

File: src/main/resources/init_db.sql (hunk 10 above)
Comment (Sashk52) [reply src/main/resources/init_db.sql]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 9 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java:20]: done

File: src/main/resources/init_db.sql (hunk 10 above)
Comment (Sashk52) [reply src/main/resources/init_db.sql:12]: done

File: .idea/compiler.xml (hunk 8 above)
Comment (Sashk52) [reply .idea/compiler.xml]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+            }
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't getAll manufacturers ", ex);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturer SET manufacturer_name=?, manufacturer_country=?"
```
Comment (boroda4436) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: Does it work?

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+import taxi.dao.ManufacturerDao;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String sqlQuery = "INSERT INTO manufacturer (name, country)"
```
Comment (boroda4436) [inline src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java:20]: Have tried to run your application? Does it work well?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 12 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java:20]: i rollback to previous version (manufacturer_name, manufacturer_country)

File: src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java (hunk 11 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java]: yes

**Review by boroda4436 — CHANGES_REQUESTED** [review]
columns naming

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**